File: engine/transpile/src/test/resources/golden/SYK008/WS_作業項目.py
"""SYK008 のデータ部レコード WS-作業項目 を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class WS_作業項目:
    """01 WS-作業項目 総バイト長 1。"""

    LENGTH = 1

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(1)
        else:
            self._data = bytearray(data)
            if len(self._data) < 1:
                self._data.extend(b"\x00" * (1 - len(self._data)))

    @property
    def data(self):
        return self._data

    # WS-検査結果
    def get_WS_検査結果(self):
        return decode_alphanumeric(self._data, 0, 1)
    def set_WS_検査結果(self, value):
        encode_alphanumeric(self._data, 0, 1, value)

    def is_WS_検査エラー(self):
        return self.get_WS_検査結果() == "E"

File: engine/transpile/src/test/resources/golden/SYK008/WS_作業項目.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK008 のデータ部レコード WS-作業項目 を逐語対訳した自動生成コード。 */
public final class WS_作業項目 {

    /** 01 WS-作業項目 総バイト長 1。 */
    public static final int LENGTH = 1;

    private final byte[] data;

    public WS_作業項目() {
        this.data = new byte[1];
    }

    public WS_作業項目(byte[] source) {
        this.data = Arrays.copyOf(source, 1);
    }

    public byte[] data() {
        return data;
    }

    // WS-検査結果
    public String get_WS_検査結果() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 1);
    }
    public void set_WS_検査結果(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 1, value);
    }

    public boolean is_WS_検査エラー() {
        return get_WS_検査結果().equals("E");
    }
}
